mirgecom/initializers.py

```diff
@@ -370,8 +468,77 @@ def exact_rhs(self, discr, q, t=0.0):
         return flat_obj_array(massrhs, energyrhs, momrhs)
 
 
+class AcousticPulse:
+    r"""Solution initializer for N-dimensional Gaussian acoustic pulse.
+
+    The Gaussian pulse is defined by:
+
+    .. math::
+
+        {\rho}E(\vec{r}) = {\rho}E + a_0 * G(\vec{r})\\
+        G(\vec{r}) = \exp^{-(\frac{(\vec{r}-\vec{r_0})}{\sqrt{2}w})^{2}},
+
+    where :math:`\vec{r}` are the nodal coordinates, and :math:`\vec{r_0}`,
+    :math:`amplitude`, and :math:`w`, are the the user-specified pulse location,
+    amplitude, and width, respectively.
+
+    .. automethod:: __init__
+    .. automethod:: __call__
+    """
+
+    def __init__(self, numdim=1, amplitude=1,
+                 center=None, width=1):
+        r"""
+        Initialize acoustic pulse parameters.
+
+        Parameters
+        ----------
+        numdim: int
+            specify the number of dimensions for the pulse
+        amplitude: float
+            specifies the value of :math:`amplitude`
+        width: float
+            specifies the rms width of the pulse
+        center: numpy.ndarray
+            pulse location, shape ``(numdim,)``
+        """
+        if len(center) == numdim:
+            self._center = center
+        elif len(center) > numdim:
+            numdim = len(center)
+            self._center = center
+        else:
+            self._center = np.zeros(shape=(numdim,))
+
+        assert len(self._center) == numdim
+
+        self._amp = amplitude
+        self._width = width
+        self._dim = numdim
+
+    def __call__(self, x_vec, q, eos=IdealSingleGas()):
+        """
+        Create the acoustic pulse at locations *x_vec*.
+
+        Parameters
+        ----------
+        t: float
+            Current time at which the solution is desired (unused)
+        x_vec: numpy.ndarray
+            Nodal coordinates
+        eos: :class:`mirgecom.eos.GasEOS`
+            Equation of state class to be used in construction of soln (unused)
+        """
+        assert len(x_vec) == self._dim
+        cv = split_conserved(self._dim, q)
+        pulse_energy = cv.energy + _make_pulse(amp=self._amp, w=self._width,
+                                               r0=self._center, r=x_vec)
+        return join_conserved(dim=self._dim, mass=cv.mass, energy=pulse_energy,
+                              momentum=cv.momentum)
```

Comment:
```suggestion
        cv = split_conserved(self._dim, q)
        return cv.replace(
            energy=cv.energy + _make_pulse(...)).join()
```
would read more nicely to me. `.replace()` is easy to implement with `dataclasses.replace` (just forward the kwargs). `.join()` (also vaporware) might be a useful method to add to the conserved variables dataclass.

Reply:
I just don't follow this. The new expression is (for me) much hard to read and decipher, and requires implementation of new methods besides.   Any chance we could defer this particular change?